### PR TITLE
Implement GitHub Actions

### LINKED ISSUE #79: Add GitHub Action for Continuous Integration

### PR DESCRIPTION
**Pull request by MarcoDotIO**
Closes #79

### CODE REVIEW

**Comment by kPatch** [comment]
The README `Actions` tag needs to be updated to point to the `aptos-labs/Aptos-Unity-SDK`.

**Review by rustielin — COMMENTED** [review]
Nice! leaving some comments so we can iterate, but overall looks good

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+          path: Library
+          key: Library-${{ hashFiles('Assets/**', 'Packages/**', 'ProjectSettings/**') }}
+          restore-keys: |
+            Library-
+
+      # Test
+      - name: Run tests
+        uses: game-ci/unity-test-runner@v2
+        env:
+          UNITY_LICENSE: ${{ secrets.UNITY_LICENSE }}
```
[the remark this answers is not in the document]
Comment (gregnazario) [reply .github/workflows/main.yaml:28]: Yeah, we can get a license to set this up

File: .github/workflows/main.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+name: Actions
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    name: Build my project
```
[the remark this answers is not in the document]
Comment (kPatch) [reply .github/workflows/main.yaml:7]: Ditto.

File: .github/workflows/main.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      - uses: actions/cache@v3
+        with:
+          path: Library
+          key: Library-${{ hashFiles('Assets/**', 'Packages/**', 'ProjectSettings/**') }}
+          restore-keys: |
+            Library-
+
+      # Test
+      - name: Run tests
+        uses: game-ci/unity-test-runner@v2
```
[the remark this answers is not in the document]
Comment (kPatch) [reply .github/workflows/main.yaml:26]: Let me double check on this with @MarcoDotIO on this.

File: .github/workflows/main.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (kPatch) [reply .github/workflows/main.yaml:28]: Let's run it with the free / personal license first, and then use the enterprise license once it works with the free license. Unity updated their verification method and running the Test Runner requires following some steps listed only the Unity-Test-Runner Discord.

**Pull request closed without merging**